### PR TITLE
Git CI setup

### PR DESCRIPTION
**Pull request by ajit-shrivastav-nyu**

### CODE REVIEW

**Comment by ajit-shrivastav-nyu** [comment]
@rpyzh @ss14635  I have added artifact attestation for the docker published build using docker push action (with sbom tag). This internally uses github atrifact attestation. The build attested can be verified.. I am attaching a text output of the build verification for the attestations uploaded
[pastebin_artifact_attestation_sbom_verification.txt](https://github.com/user-attachments/files/17368538/pastebin_artifact_attestation_sbom_verification.txt)

File: app_test.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    def test_add_note_without_token(self):
+        response = self.app.post('/add_note')
+        self.assertEqual(response.status_code, 403) 
+        
+    def test_delete_note_without_token(self):
+        response = self.app.post('/delete_note')
+        self.assertEqual(response.status_code, 403) 
+
+    def test_update_note_without_token(self):
+        response = self.app.post('/delete_note')
```
Comment (ss14635) [inline app_test.py:45]: Should be '/update_note'

File: main.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 
 users = {}
 notes = {}
-secret_key = os.environ["PASTEBIN_JWT_SECRET"]
+secret_key = 'secret'#os.environ["PASTEBIN_JWT_SECRET"]
```
Comment (ss14635) [inline main.py:16]: Should 'secret' be hard-coded here?

**Review by ss14635 — CHANGES_REQUESTED** [review]
Left some in-line comments which should be reviewed and closed. Overall looks good to me. Thanks!

File: main.py (hunk 2 above)
Comment (ss14635) [reply main.py:16]: I assume we have this to see if the secret scanning works properly?

File: app_test.py (hunk 1 above)
Comment (ajit-shrivastav-nyu) [reply app_test.py:45]: yes, copy paste error. let me fix

File: main.py (hunk 2 above)
Comment (ajit-shrivastav-nyu) [reply main.py:16]: yes, was testing this and trufflehog not detecting it.. ideally this would go in github secrets (codespaces)

Comment (ss14635) [reply main.py:16]: Do we want to revert it in this PR or leave it for further testing with trufflehog?

File: .github/workflows/push_build.yml [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+env:
+  IMAGE_NAME: ajitshrivastav/pastebin
+
+jobs:
+  publish_to_docker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Env
+        env:
+          DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
```
Comment (ss14635) [inline .github/workflows/push_build.yml:13]: Security hardening check: Use of Github secret in workflow - https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions#using-secrets-in-a-workflow

File: main.py (hunk 2 above)
Comment (ajit-shrivastav-nyu) [reply main.py:16]: i think we can revert it.. we can use a new variable with secret if we want to continue testing trufflelog

Comment (ss14635) [reply main.py:16]: Sounds good. Lets do that.

Comment (ss14635) [inline main.py:16]: In order for Trufflehog to detect this as a Github secret, change the variable name to something like `github_secret` and make the value a SHA256 hash like `123963c1434c377428ca8531fbc46c0c43d037a1`

Comment (ss14635) [reply main.py:16]: As discussed in Slack, update this to something that Trufflehog will catch as an unverified secret.

Comment (ajit-shrivastav-nyu) [reply main.py]: This is taken care. Secret is coming from CodeSpaces secret

**Review by ss14635 — APPROVED** [review]
New changes since my prior review look fine.

**Review by rpyzh — APPROVED** [review]

**Comment by rpyzh** [comment]
Unverified commits were confirmed with Ajit via Slack (reason: temporary issue with signing key). Moreover, unverified commits contain trivial (whitespace) changes and are not malicious. Will proceed with merging with branch protection bypass.

**Pull request merged**